python/tests/arithmetic_tests/test_date.py

```diff
@@ -1,135 +1,684 @@
+                      CAST((minutes) AS VARCHAR) AS min_res,
+                      CAST((hours) AS VARCHAR) AS hrs_res,
+                      CAST((days) AS VARCHAR) AS days_res,
+                      CAST((months) AS VARCHAR) AS mths_res,
+                      CAST((years) AS VARCHAR) AS yrs_res
+                      FROM adate_minus_date"""
 
 
 # Equivalent SQL for MySQL
 # SELECT
 # 	id,
-#     TIMESTAMPDIFF(SECOND, c2, c1) AS f_c1,
-#     TIMESTAMPDIFF(DAY, c2, c1) AS f_c2,
-#     TIMESTAMPDIFF(MONTH, c2, c1) AS f_c3
+#     TIMESTAMPDIFF(SECOND, c2, c1) AS seconds,
+#     TIMESTAMPDIFF(MINUTE, c2, c1) AS minutes,
+#     TIMESTAMPDIFF(HOUR, c2, c1) AS hours,
+#     TIMESTAMPDIFF(DAY, c2, c1) AS days,
+#     TIMESTAMPDIFF(MONTH, c2, c1) AS months,
+#     TIMESTAMPDIFF(YEAR, c2, c1) AS years
 # FROM date_tbl;
 
 
-class arithtst_date_minus_interval(TstView):
+# Using explicit interval types for subtraction and addition
+class arithtst_date_minus_sinterval(TstView):
     def __init__(self):
         # Validated on MySQL
+        # The subtraction of all interval types produces the same result as subtracting MINUTE type in MySQL
         self.data = [
             {
                 "id": 0,
                 "seconds_c1": "2014-11-05",
+                "minutes_c1": "2014-11-05",
+                "hours_c1": "2014-11-05",
                 "days_c1": "2014-11-05",
-                "months_c1": "2014-11-05",
             },
             {
                 "id": 1,
                 "seconds_c1": "2023-02-26",
+                "minutes_c1": "2023-02-26",
+                "hours_c1": "2023-02-26",
                 "days_c1": "2023-02-26",
-                "months_c1": "2023-02-21",
             },
             {
                 "id": 2,
                 "seconds_c1": "1948-12-02",
+                "minutes_c1": "1948-12-02",
+                "hours_c1": "1948-12-02",
                 "days_c1": "1948-12-02",
-                "months_c1": "1948-12-21",
             },
         ]
-        self.sql = """CREATE MATERIALIZED VIEW date_minus_interval AS SELECT
+        self.sql = """CREATE MATERIALIZED VIEW date_minus_sinterval AS SELECT
                       v1.id,
                       c1 - (v1.seconds) AS seconds_c1,
-                      c1 - (v1.days) AS days_c1,
-                      c1 - (v1.months) AS months_c1
-                      FROM date_minus_date v1
+                      c1 - (v1.minutes) AS minutes_c1,
+                      c1 - (v1.hours) AS hours_c1,
+                      c1 - (v1.days) AS days_c1
+                      FROM adate_minus_date v1
                       JOIN date_tbl v2 ON v1.id = v2.id;"""
 
 
-class arithtst_date_plus_interval(TstView):
+class arithtst_date_minus_linterval(TstView):
+    def __init__(self):
+        # Validated on MySQL
+        # The result of subtracting MONTH type interval matches with MySQL
+        # whereas YEARS behaves similarly as MONTHS in terms of accuracy
```

Comment:
if the result is different from mysql please file an issue so I can look into it.